NBitcoin/Protocol/NodesCollection.cs

```diff
@@ -64,6 +64,25 @@ public void PushMessage(IncomingMessage message)
 
 			#endregion
 		}
+
+		/// <summary>
+		/// Provides a comparer to specify how nodes are compared for equality
+		/// </summary>
+		public class NodeComparer:IEqualityComparer<Node>
+		{
+			public bool Equals(Node a, Node b)
+			{
+				bool same = a.RemoteSocketAddress.MapToIPv6().ToString() == b.RemoteSocketAddress.MapToIPv6().ToString() && a.RemoteSocketPort == b.RemoteSocketPort;
```

Comment:
We should probably add some checks to see that a and b are not null.

Reply:
Done